packages/blocks/src/components/remote-selection/remote-selection.ts

```diff
@@ -159,6 +163,31 @@ export class RemoteSelection extends LitElement {
             });
             return;
           }
+          if (
+            models.length === 1 &&
+            matchFlavours(models[0], ['affine:database'] as const)
+          ) {
+            restoreSelection({
+              type: 'DatabaseTitle',
+              startOffset: userRange.startOffset,
+              endOffset: userRange.endOffset,
+              models: [models[0]],
+            });
+            return;
+          }
+          if (
+            models.length === 2 &&
+            matchFlavours(models[1], ['affine:database'] as const)
+          ) {
+            restoreSelection({
+              type: 'DatabaseRichText',
+              startOffset: userRange.startOffset,
+              endOffset: userRange.endOffset,
+              models: [models[0], models[1]],
+              databaseCellPos: userRange.blockPos,
```

Comment:
I don't think it's good if adding new block type requires modifying the cursor selection module. @Flrande do we have some solution to avoid this kind of code smell?

In the worst case, I think ignoring cell cursor in database block in the first version can still be acceptable, which is still better than hard-coded here. For the non-title columns in database block, they don't have to support cursor.

Reply:
Considering that there will be a lot of non-text input scenarios (such as dates, statuses, progress bars, etc.) in the database, I think we can ignore the internal selection tracking.